naga/src/back/spv/block.rs

```diff
@@ -253,6 +258,445 @@ impl Writer {
                 _ => {}
             }
         }
+        // OpEmitMeshTasksEXT must be called right before exiting (after setting other
```

Comment:
This function in general seems like a good candidate to be split up into many different functions to make things easier to follow along. They seem decently self-contained.